pkg/virt-handler/device-manager/socket_device.go

```diff
@@ -56,12 +53,16 @@ func NewSocketDevicePlugin(socketName, socketDir, socket string, maxDevices int)
 			resourceName: fmt.Sprintf("%s/%s", DeviceNamespace, socketName),
 			initialized:  false,
 			lock:         &sync.Mutex{},
+			done:         make(chan struct{}),
+			deregistered: make(chan struct{}),
+			socketPath:   SocketPath(strings.Replace(socketName, "/", "-", -1)),
 		},
-		pluginSocketPath: SocketPath(strings.Replace(socketName, "/", "-", -1)),
-		socket:           socket,
-		socketDir:        socketDir,
-		socketName:       socketName,
+		socket:     socket,
+		socketDir:  socketDir,
+		socketName: socketName,
 	}
+	dpi.registerFunc = dpi.registerServer
```

Comment:
You can move this inside the DevicePluginBase declaration as suggested below in https://github.com/kubevirt/kubevirt/pull/11344/commits/44f04391b09be73f4fa2176c426e8045ec64774a#r1592508779